app/src/main/java/com/naman14/timberx/ui/viewmodels/MainViewModel.kt

```diff
@@ -183,27 +186,15 @@ class MainViewModel(
         }
 
         override fun addToPlaylist(song: Song) {
-            AddToPlaylistDialog.newInstance(song).show((context as AppCompatActivity).supportFragmentManager, "AddPlaylist")
+            AddToPlaylistDialog.show(context as AppCompatActivity, song)
```

Comment:
An Activity context should not be inside ViewModel. This will leak across config change.